com.palantir.typescript/src/com/palantir/typescript/preferences/ProjectPreferenceStore.java

```diff
@@ -202,6 +251,186 @@ private synchronized void insertValue(String name) {
+                value = value.toUpperCase();
+                if (TSCONFIG_MODULE_RESOLUTION_MAPPING.containsKey(value)) {
+                    value = TSCONFIG_MODULE_RESOLUTION_MAPPING.get(value);
+                }
+            } else if (matchingPreference.equals(IPreferenceConstants.COMPILER_JSX)) {
+                value = value.toUpperCase();
+            }
+        }
+        return value;
+    }
+
+    private boolean isTsConfigPreference(String name) {
+        return PREFERENCE_TO_TSCONFIG_PATH.containsKey(name);
+    }
+
+    private boolean isSupportedTsConfigPath(String tsConfigJsonPath) {
+        return TSCONFIG_PATH_TO_PREFERENCE.containsKey(tsConfigJsonPath);
+    }
+
+    private String getFileSHA1(IFile file) {
+
+        if (!file.exists()) {
+            return null;
+        }
+
+        InputStream fileInputStream = null;
+        try {
+            fileInputStream = file.getContents();
+            MessageDigest md = MessageDigest.getInstance("SHA1");
+            byte[] dataBytes = new byte[1024];
+
+            int nread = 0;
+
+            while ((nread = fileInputStream.read(dataBytes)) != -1) {
+                md.update(dataBytes, 0, nread);
+            }
+
+            byte[] mdbytes = md.digest();
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < mdbytes.length; i++) {
+                sb.append(Integer.toString((mdbytes[i] & 0xff) + 0x100, 16).substring(1));
+            }
+            return sb.toString();
+        } catch (Exception e) {
+            String errorMessage = "Cannot sha1 '" + file + "'";
+            Status status = new Status(IStatus.ERROR, TypeScriptPlugin.ID, errorMessage, e);
+            TypeScriptPlugin.getDefault().getLog().log(status);
+            return null;
+        } finally {
+            try {
+                if (fileInputStream != null) {
+                    fileInputStream.close();
+                }
+            } catch (Exception e) {
+            }
+        }
+    }
+
+    private boolean didTsConfigFileChanged() {
+        IEclipsePreferences projectPreferences = this.getProjectPreferences();
+
+        IFile tsConfigFile = getTsConfigFile();
+        boolean changed = true;
+        if (tsConfigFile.exists()) {
+            changed = false;
+            try {
+                long previousLastModTimestamp = projectPreferences
+                    .getLong(IPreferenceConstants.PREFERENCE_STORE_TS_CONFIG_LAST_MODIFICATION_TIME, -1);
+                long currentLastModTimestamp = tsConfigFile.getModificationStamp();
+                if (previousLastModTimestamp != currentLastModTimestamp) {
+
+                    // optimization: get hash only if mod timestamp changed
+                    String previousHash = projectPreferences.get(IPreferenceConstants.PREFERENCE_STORE_TS_CONFIG_HASH, null);
+                    String currentHash = getFileSHA1(tsConfigFile);
+                    if (!Objects.equals(previousHash, currentHash)) {
```

Comment:
actually, project's java compiler version is set to 6. I think my Eclipse did not complain because I have no JDK < 7 installed on my workstation, I will fix it.